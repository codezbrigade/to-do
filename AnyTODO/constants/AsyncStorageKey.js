export const todoKey = 'com.codezbrigade.todo';
export const expoPushTokenKey = 'expoPushToken';
export const ratingKey = 'com.codezbrigade.todo.rating';
export const countKey = 'com.codezbrigade.todo.launchCount';
