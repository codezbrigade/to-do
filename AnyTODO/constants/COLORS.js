export const COLORS = {
  main: '#2E3192',
  statusBar: 'rgba(0,0,0,0)',
  university: '#ffe500',
  work: '#00b2ff',
  grocery: '#a8ff26',
  customize: '#1effad',
  home: '#FF9680',
  white: '#fff',
  modalText: '#7F7F7F',
  // api: '#f2f2f2',
  // apiFont: '#525252',
  factBg: '#ffffcc',
  factBorder: '#ffff00',
  greetings: '#303030',
  time: '#B9B9B9',
  sub_title: 'rgba(0,0,0,0.5)',
  modalBackground: 'rgba(0,0,0,0.25)',
  searchBar: '#f1f1f1',
  inActiveHeader: '#484848',
  previewContainerBorder: '#bfbfbf',
  previewBg: 'rgba(253, 252, 252, 0.95)',
  grey_b: '#767676',
  grey_c: '#979797',
  delete: '#d12f2f',
  black: '#000',
  black_1: 'rgba(0,0,0,0.87)'
};

