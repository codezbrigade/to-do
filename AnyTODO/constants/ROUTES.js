export const ROUTES = {
  home_screen: 'Home',
  new_task_screen: 'NewTask',
  auth_screen: 'Auth'
};