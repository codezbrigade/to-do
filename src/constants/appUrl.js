export const appUrl =
  'https://play.google.com/store/apps/details?id=com.halfbrick.fruitninjafree';
