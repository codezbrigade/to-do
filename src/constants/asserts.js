import addTask from '../asserts/images/addTask.png';
import calenderLogo from '../asserts/images/calender-Logo.png';
import close from '../asserts/images/close.png';
import university from '../asserts/images/univesiry.png';
import grocery from '../asserts/images/grocery.png';
import work from '../asserts/images/work.png';
import customize from '../asserts/images/customize.png';
import repeat from '../asserts/images/repeat.png';
import selected from '../asserts/images/selected.png';
import home from '../asserts/images/home.png';
import down from '../asserts/images/down.png';
import calender from '../asserts/images/calender.png';
import deleteIcon from '../asserts/images/delete.png';
import edit from '../asserts/images/edit.png';
import bgimg from '../asserts/images/background.jpg';
import plus from '../asserts/images/plus.png';
import search from '../asserts/images/search.png';
import closeBlack from '../asserts/images/closeBlack.png';
import searchLogo from '../asserts/images/searchLogo.png';
import closeBtn from '../asserts/images/closeBtn.png';
import checked from '../asserts/images/checked.png';
import searchCloseLogo from '../asserts/images/searchCloseLogo.png';
import water from '../asserts/images/water.png';
import rateHeadLogo from '../asserts/images/rateHeadLogo.png';
import ratedStar from '../asserts/images/rated-star.png';
import unratedStar from '../asserts/images/unrated-star.png';
import goback from '../asserts/images/goback.png';
import clock from '../asserts/images/clock.png';
import description from '../asserts/images/description.png';
import logout from '../asserts/images/logout.png';

export const asserts = {
  logout,
  ratedStar,
  clock,
  description,
  goback,
  unratedStar,
  rateHeadLogo,
  searchCloseLogo,
  water,
  closeBtn,
  checked,
  searchLogo,
  closeBlack,
  search,
  plus,
  bgimg,
  addTask,
  calenderLogo,
  close,
  university,
  grocery,
  work,
  customize,
  repeat,
  selected,
  home,
  down,
  calender,
  deleteIcon,
  edit,
};
