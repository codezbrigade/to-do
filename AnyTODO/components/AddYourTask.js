import React, { useLayoutEffect, useRef } from 'react';
import { Animated, Image, Pressable, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { COLORS, asserts, FONTS, strings } from '../constants';

import image01 from '../asserts/images/image01.png';

import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';


const AddYourTask = ({ handlePress, selectedHeader }) => {
  const opacity = useRef(new Animated.Value(0)).current;

  let str = selectedHeader !== strings.completed ? strings.add_your_task : strings.no_task_completed;

  useLayoutEffect(() => {
    if (opacity) opacity.setValue(0);

    Animated.timing(opacity, {
      toValue: 1,
      useNativeDriver: false
    }).start();

    // LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
  }, [selectedHeader]);

  if (selectedHeader !== strings.completed) {
    return (
      <Animated.View style={{ opacity }}>
        <Pressable
          dataTest='others'
          onPress={handlePress}
          style={({ pressed }) => [styles.logoCotainer, pressed && styles.opacity]}
        >
          <Image source={image01} style={styles.image01} />
          <Text dataTest='others' style={styles.text1}>{strings.add_your_task}</Text>
        </Pressable>
      </Animated.View>
    );
  }

  return (
    <Animated.View style={[styles.container, { opacity }]}>
      <View style={styles.textContainer}>
        <Text dataTest='completed' style={styles.text}>{str}</Text>
      </View>
      <View style={{ width: wp(3.6) }}>
        {
          selectedHeader !== strings.completed && <TouchableOpacity onPress={handlePress}>
            <Image source={asserts.plus} style={styles.img} />
          </TouchableOpacity>
        }
      </View>
    </Animated.View>
  );
};

export default AddYourTask;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: hp(6.64),
    backgroundColor: COLORS.white,
    borderRadius: 16,
    // elevation: 3,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  img: {
    height: 15,
    width: 15
  },
  text: {
    // textAlign: 'center',
    fontSize: 16,
    fontWeight: '600',
    fontFamily: FONTS.LatoRegular
  },
  textContainer: {
    width: '65%',
    paddingLeft: '2%'
  },
  image01: {
    height: 190,
    width: 215,
    resizeMode: 'contain',
  },
  text1: {
    textAlign: 'center',
    position: 'absolute',
    bottom: '3%',
    fontFamily: FONTS.RobotoRegular_400,
    fontSize: 12,
    lineHeight: 14.06,
    color: COLORS.white,
    alignSelf: 'center',
  },
  logoCotainer: {
    alignSelf: 'center',
  },
  opacity: {
    opacity: 0.6
  }
});
