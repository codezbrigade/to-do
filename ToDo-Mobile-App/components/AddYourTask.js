import React from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { COLORS, asserts, FONTS, strings } from '../constants';

import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';


const AddYourTask = ({ handlePress, selectedHeader }) => {

  let str = selectedHeader !== strings.completed ? strings.add_your_task : strings.no_task_completed;

  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <Text style={styles.text}>{str}</Text>
      </View>
      <View style={{ width: wp(3.6) }}>
        {
          selectedHeader !== strings.completed && <TouchableOpacity onPress={handlePress}>
            <Image source={asserts.plus} style={styles.img} />
          </TouchableOpacity>
        }
      </View>
    </View>
  );
};

export default AddYourTask;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: hp(6.64),
    backgroundColor: COLORS.white,
    borderRadius: 16,
    // elevation: 3,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  img: {
    height: 15,
    width: 15
  },
  text: {
    // textAlign: 'center',
    fontSize: 16,
    fontWeight: '600',
    fontFamily: FONTS.LatoRegular
  },
  textContainer: {
    width: '65%',
    paddingLeft: '2%'
  }
})