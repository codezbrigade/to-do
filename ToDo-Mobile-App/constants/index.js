import { strings } from './localize';

import { COLORS } from './COLORS';
import { FONTS } from './FONTS';

import { asserts } from './asserts';
import { ROUTES } from './ROUTES';
import {
  todoKey,
  expoPushTokenKey,
  ratingKey,
  countKey
} from './AsyncStorageKey';

const categories = [
  {
    name: strings.university,
    color: COLORS.university
  },
  {
    name: strings.work,
    color: COLORS.work
  },
  {
    name: strings.grocery,
    color: COLORS.grocery
  },
  {
    name: strings.home,
    color: COLORS.home
  },
  {
    name: strings.customize,
    color: COLORS.customize
  }
];

const categoryLogoMap = {
  [strings.grocery]: asserts.grocery,
  [strings.work]: asserts.work,
  [strings.university]: asserts.university,
  [strings.customize]: asserts.customize,
  [strings.home]: asserts.home,
};

const HEADERS = [{ title: strings.today, },
{ title: strings.tomorrow },
{ title: strings.upcoming },
{ title: strings.completed }
]

const MONTHS = [
  strings.Jan, strings.Feb, strings.Mar, strings.Apr, strings.May,
  strings.Jun, strings.Jul, strings.Aug, strings.Sep, strings.Oct,
  strings.Nov, strings.Dec
];

export {
  asserts,
  FONTS,
  COLORS,
  strings,
  categories,
  categoryLogoMap,
  todoKey,
  expoPushTokenKey,
  ratingKey,
  countKey,
  HEADERS,
  MONTHS,
  ROUTES
}