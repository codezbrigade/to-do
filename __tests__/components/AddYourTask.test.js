import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AddYourTask from '../../AnyTODO/components/AddYourTask';
import { strings } from '../../AnyTODO/constants';

const setup = (props = {}, state = null) => {
  const wrapper = shallow(<AddYourTask {...props} />);
  if (state) wrapper.setState(state);
  return wrapper;
};

const findTestAttr = (wrapper, attr) =>
  wrapper.find(`[dataTest='${attr}']`);

const props = {
  selectedHeader: strings.completed
};

describe('AddYourTask component basics', () => {
  test('if selectedHeader prop is not completed', () => {
    const wrapper = setup();
    const appComponent = findTestAttr(wrapper, 'others');
    expect(appComponent.length).toBe(2);
  });
  test('if selectedHeader prop is Completed', () => {
    const wrapper = setup(props);
    const appComponent = findTestAttr(wrapper, 'completed');
    expect(appComponent.length).toBe(1);
  });
  test('Snapshot testing', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});