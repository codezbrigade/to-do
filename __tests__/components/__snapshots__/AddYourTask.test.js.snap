// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddYourTask component basics Snapshot testing 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    {
      "opacity": 0,
    }
  }
>
  <Pressable
    dataTest="others"
    style={[Function]}
  >
    <Image
      source={
        {
          "testUri": "../../../AnyTODO/asserts/images/image01.png",
        }
      }
      style={
        {
          "height": 190,
          "resizeMode": "contain",
          "width": 215,
        }
      }
    />
    <Text
      dataTest="others"
      style={
        {
          "alignSelf": "center",
          "bottom": "3%",
          "color": "#fff",
          "fontFamily": "RobotoRegular",
          "fontSize": 12,
          "lineHeight": 14.06,
          "position": "absolute",
          "textAlign": "center",
        }
      }
    >
      Add your task
    </Text>
  </Pressable>
</ForwardRef(AnimatedComponentWrapper)>
`;
